nima/http2/webserver/src/main/java/io/helidon/nima/http2/webserver/Http2Connection.java

```diff
@@ -240,11 +266,40 @@ private void doHandle() throws InterruptedException {
                     // no data to read -> connection is closed
                     throw new CloseConnectionException("Connection closed by client", e);
                 }
+                dispatchHandler();
+
+                // Flow-control: frame processing is done, free space reserved for frame in window
+                int length = frameHeader.length();
+                if (length > 0) {
+                    int streamId = frameHeader.streamId();
+                    if (streamId > 0 && frameHeader.type() != Http2FrameType.HEADERS) {
+                        // Stream ID > 0: update conenction and stream
+                        FlowControl.Inbound inboundFlowControl = stream(streamId)
+                                .stream()
+                                .inboundFlowControl();
+                        inboundFlowControl.incrementWindowSize(length);
+                        LOGGER.log(System.Logger.Level.INFO, () -> String.format(
+                                "SRV IFC: Full increment %d-> %d", length,
+                                inboundFlowControl.getRemainingWindowSize()));
+                    } else {
+                        // Stream ID == 0: update connection only
+                        inboundWindowSize.incrementWindowSize(length);
+                        LOGGER.log(System.Logger.Level.INFO, () -> String.format(
+                                "SRV IFC: Conn increment %d -> %d",
+                                length, inboundWindowSize.getRemainingWindowSize()));
+                    }
+                }
```

Comment:
Increment should be just oneliner in `io.helidon.nima.http2.webserver.Http2Stream#readEntityFromPipeline`